radix-engine/src/system/system_modules/execution_trace/module.rs

```diff
@@ -47,6 +47,28 @@ impl ExecutionTraceModule {
     }
 }
 
+/// Metrics gathered during transaction execution.
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor, Default)]
+pub struct ExecutionMetrics {
+    /// Consumed cost units (excluding royalties)
```

Comment:
Should probabaly move this to somewhere else. For typical validators, `execution_trace` is turned off but we  need `execution_metrics` for consensus.

Reply:
Moved to `transaction_receipt.rs`.